config/processors/event_hub_log_security_azure.event_hub.conf

```
@@ -15,13 +15,64 @@ filter {
     add_field => { "observer.type" => "eventhub" }
   }
   mutate {
-    add_field => { 
-      "event.module" => "Azure Security Logs"
-      }
     copy => { "message" => "log.original" }
   }
+  mutate {
+    gsub => ["message" ," ",""]
+  }
+  json {
+    source => "message"
+    target => "tmp"
+  }
+
+  split {
+    field => "[tmp][records]"
+    target => "tcs"
+  }
+
```

Original message field has list of events. After the split, each event would have original message as list of all events which is not needed. After the split please add

```suggestion
 json_encode {
    source => "tcs"
    target => "message"
  }
```

We have done the same in azure audit and okta.
@brian-grabau Do you agree?

---

Yes that is probably true, split would split into separate messages